src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1568,6 +1598,38 @@ getTestNameDescriptions(bool addAdditionOptions)
   return reportDescriptionCollection;
 }
 
+std::string
+extendedKeysOptions()
+{
+  static unsigned int m_maxColumnWidth = 100;
+  std::stringstream fmt_output;
+  // Formatting color parameters
+  // Color references: https://en.wikipedia.org/wiki/ANSI_escape_code
+  const std::string fgc_header     = XBU::is_escape_codes_disabled() ? "" : EscapeCodes::fgcolor(EscapeCodes::FGC_HEADER).string();
+  const std::string fgc_optionName = XBU::is_escape_codes_disabled() ? "" : EscapeCodes::fgcolor(EscapeCodes::FGC_OPTION).string();
+  const std::string fgc_optionBody = XBU::is_escape_codes_disabled() ? "" : EscapeCodes::fgcolor(EscapeCodes::FGC_OPTION_BODY).string();
+  const std::string fgc_reset      = XBU::is_escape_codes_disabled() ? "" : EscapeCodes::fgcolor::reset();
+
+  // Report option group name (if defined)
+  boost::format fmtHeader(fgc_header + "\n%s:\n" + fgc_reset);
+    fmt_output << fmtHeader % "EXTENDED KEYS";
+
+  static std::map<std::string, std::string> test_to_parameter = {
```

Comment:
Shouldn't this be a vector pair or multi-map (which is a pain to use). 

I ask for what if the "dma" validate has more then one parameter.

Reply:
Now you are encountering an issue with validate which I was afraid of.   

When this code was created, each test wasn't encapsulated in their own class.  If this was the case, adding support here would be relatively straight forward.  But because that is NOT the case, the SubCmdValidate needs to maintain and orchestrate this parameters.   In addition, it is making the code difficult to support.

Reply:
Maybe it would be worth to move the test case for dma into its own suboption in validate?